python/dlisio/ext/core.cpp

```diff
@@ -371,26 +371,28 @@ noexcept (false) {
                 // const auto msg = 'Non-sequential frames. expected = {}, current = {}'
             }
 
-            const auto tail = std::distance(ptr, end);
-            auto size_src = pre_size.src + data_size.src + post_size.src;
-            if (tail < size_src) {
-                const auto msg = "unaligned record: tail (which is "
-                               + std::to_string(tail)
-                               + ") < fmt_size (which is "
-                               + std::to_string(size_src)
-                               + ")"
-                               ;
-                throw std::runtime_error(msg);
-            }
+            auto assert_overflow = [end](const char* ptr, int skip) {
+                if (ptr + skip > end) {
+                    const auto msg = "corrupted record: fmtstr would read past end";
+                    throw std::runtime_error(msg);
+                }
+            };
 
-            ptr += pre_size.src;
+            int src_skip, dst_skip;
```

Comment:
Note: 'skip' sounds a bit weird as you use the same variable to actually 'skip' data and read data. Naming got me quite confused.